src/devices/CpuTemperature/samples/CpuTemperature.Sample.cs

```diff
@@ -13,17 +13,29 @@ internal class Program
         public static void Main(string[] args)
         {
             CpuTemperature cpuTemperature = new CpuTemperature();
+            Console.WriteLine("Press any key to quit");
 
-            while (true)
+            while (!Console.KeyAvailable)
             {
                 if (cpuTemperature.IsAvailable)
                 {
-                    double temperature = cpuTemperature.Temperature.DegreesCelsius;
-                    if (!double.IsNaN(temperature))
+                    var temperature = cpuTemperature.ReadTemperatures();

```

Comment:
I'd recommend to leave the old API alone and throw if there are no thermometrs/data and add new API on top of it (on Linux enumerate just 1)

Reply:
I think I left the old API in place, and just updated the test to use the new one, but I'll check.

Reply:
sounds good in that case